src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/ConnectivityInformation.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.connectivity;
+
+import software.amazon.awssdk.services.greengrassv2data.model.ConnectivityInfo;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ConnectivityInformation {
+    private static final Pattern IPV4_PAT = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)(?::(\\d+)){0,1}$");
+    private static final Pattern IPV6_DOUBL_COL_PAT = Pattern.compile(
+            "^\\[{0,1}([0-9a-f:]*)::([0-9a-f:]*)(?:\\]:(\\d+)){0,1}$");
+    private static String ipv6Pattern;
+    private static final Pattern IPV6_PAT = Pattern.compile(ipv6Pattern);
+
+    static {
+        ipv6Pattern = "^\\[{0,1}";
+        for (int i = 1; i <= 7; i++) {
+            ipv6Pattern += "([0-9a-f]+):";
+        }
+        ipv6Pattern += "([0-9a-f]+)(?:\\]:(\\d+)){0,1}$";
+    }
+
+    private final String hostAddress;
+
+
+    public ConnectivityInformation(String hostAddress) {
+        this.hostAddress = hostAddress;
+    }
+
+    public static ConnectivityInformation fromConnectivityInfo(ConnectivityInfo connectivityInfo) {
+        return new ConnectivityInformation(connectivityInfo.hostAddress());
+    }
+
+    // TODO: Need to implement equals method so de-dup works as intended when adding multiple
+    // connectivity information objects to a set
+
+    /**
+     * Checks if the connectivity information is a valid IPv4 or IPv6 address.
+     *
+     * @return true if valid IPv4 or IPv6 address, else false
+     */
+    public boolean isIPAddress() {
+        return isValidIP(hostAddress);
+    }
+
+    /**
+     * Checks if the connectivity information is a valid DNS name.
+     *
+     * @return false if valid IPv4 or IPv6 address, else true
+     */
+    public boolean isDNSName() {
+        return !isIPAddress();
```

Comment:
Is it safe to assume that it is a DNS name if it is not a valid IpAddress? or would we need additional information?

Reply:
There is probably some specification for what a valid DNS name looks like. Existing logic basically does "if IP address then ..., else DNS name". Can beef this up in the future if need be